html/browsers/browsing-the-web/back-forward-cache/events.html

```diff
@@ -0,0 +1,35 @@
+<!DOCTYPE HTML>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="/common/utils.js"></script>
+<script src="/html/cross-origin-embedder-policy/credentialless/resources/dispatcher.js"></script>
+<script src="resources/helper.sub.js"></script>
```

Comment:
Can we use module imports for these as much as possible? At least the libraries you are adding if there's not reason not to, it really helps to know which import a function comes from when trying to figure out the code.

Reply:
Sounds good. Things to do are:

- Create a module version of `/common/utils.js` (as this is used as a classic script in many tests, it's harder to modularize this file, e.g. need to modify wpt `serve.py`)
- Modularize `dispatcher.js`. Probably we can also move the dispatcher framework to `/common/`.

The test would look like:

```
<script type="module">
import {token} from '/common/utils.mjs';
// we'd have "import {send} from '/common/dispatcher/dispatcher.js'" if we use send() in test body.
import * as helper from './resources/helper.sub.js';

... helper.executorPath + ...
````

Reply:
IMO duplicating /common/utils.js is not worth it, and adds a maintenance burden to the commons to keep them synced. I suggest keeping that as a classic script and only working on modularizing newly-introduced files.

Reply:
> maintenance burden

Sounds reasonable too.

Just curious, are there motivations/plans for gradually modularizing scripts in WPTs in general, or are we going to keep existing classic scripts as-is for mid-term?

(`testharness.js` *somehow* works both as a classic script or a module script; it explicitly exposes methods to the global object and thus tests can use its APIs via `window` etc., not via module's import/export system even imported as a module though; This allows `any.js` tests to use `testharness.js` as a classic script in `worker` and as a module script in `moduleworker` cases, but this kind of thing isn't applicable to scripts like `utils.js`)

Reply:
I'm not aware of any plans to modularize WPT scripts; this is the first request I've heard.

Reply:
Yeah, this is just me, I really dislike flat namespaces and I'm concerned that executor.html will be a problem. Way back, I fixed the WPT server to serve .mjs files correctly but I was using it in chrome-private tests (which have since been deleted) and I don't think there are any public .mjs files.

I agree we should not duplicate existing code just for this.

Reply:
During the preparation for modularization, I found the dispatcher framework is used in workers and service workers (outside BFCache).
So modularizing these would make the worker-related COEP/COOP tests fail in browsers where module workers/module serviceworkers are not yet implemented.

So I'll keep the scripts classic for now.